documentation/en/actor-events-api.md

```diff
@@ -0,0 +1,432 @@
+
+Builtin actor events share basic similarities to the existing events emitted by user-programmed actors in FVM, but each have a specific schema that reflects their specific concerns. They also all use CBOR encoding for their values. There are also new APIs in Lotus to support querying for these events that bear some similarities to the existing FEVM `Eth*` APIs for querying events but are unique to builtin actors.
+
+## ActorEvent structure
+
+Introduced in [FIP-0049](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0049.md), events use a structured logging style of composition, containing a list of entries that define properties of the event. The log entries are described below as `EventEntry` and have the same schema for user-programmed and builtin actor events. `ActorEvent` is specifically for representing builtin actor events and includes the list of entries, the actor that emitted the event, and some metadata about the event.
+
+```ipldsch
+type ActorEvent struct {
+    entries [EventEntry] # Event entries in log form.
+    emitter Address      # Filecoin address of the actor that emitted this event.
+    # Reverted is set to true if the message that produced this event was reverted because of a
+    # network re-org in that case, the event should be considered as reverted as well.
+    reverted Bool        
+    height ChainEpoch    # Height of the tipset that contained the message that produced this event.
+    tipsetCid &Any       # CID of the tipset that contained the message that produced this event.
+    msgCid &Any          # CID of message that produced this event.
+}
+
+type EventEntry struct {
+    flags Int   # A bitmap conveying metadata or hints about this entry.
+    key String  # The key of this entry.
+    codec Int   # The value's IPLD codec.
+    value Bytes # The value of this entry as a byte string, encoded with 'codec'.
+}
+```
+
+A `flags` field is used to convey metadata or hints about the entry, currently this is used to provide an indication of the suitability of that field for indexing when stored or when used for querying.
+
+* A `flag` of `0x00` indicates that neither the key nor value are suitable for indexing.
+* A `flag` of `0x01` indicates that the key only is suitable for indexing.
+* A `flag` of `0x02` indicates that the value is suitable for indexing.
+* A `flag` of `0x03` indicates that both the key and value are suitable for indexing.
+
+Suitability for indexing is a hint, and will typically indicate the uniqueness of the content. For example if a value has an arbitrary number, such as a `piece-size`, it's unlikely to be suitable for building an index for. However, if a field such as a `"provider"` contains an actor ID, it is likely to be suitable for indexing because queries may be made for all events from a particular actor.
```

Comment:
I think the logic here is a bit off, and carries some incorrect assumptions. It's not the uniqueness/repetitiveness of a value that determines whether an index is appropriate or not, but rather its queriability. One can perfectly build an index for high variability or unique data, DBMS do this all the time. But our query APIs only offer equality predicates today, so that's what influences what we chose to tag as indexed, eg makes sense to index a CID, but it makes sense to index an arbitrary size. If we exposed gt, lt, etc predicates in queries, indexing the latter would become more worthwhile.

Reply:
@raulk OK, good input, but can you suggest new wording here please to reflect that?

```suggestion
Suitability for indexing is a hint, and will typically indicate the queriability content.
```

That's about the best I have once you boil it down. From a _user_ perspective, what do we have to say about this field? We're not even using it internally and so far have no plans to. If we can imagine users relying on it, how do we describe it? Maybe leave it simple and move on.

Reply:
I integrated this into the sentence above the flags description since it already says something similar. I removed this paragraph entirely.